packages/cosmos/src/client.ts

```diff
@@ -79,8 +91,18 @@ export class CosmosClient {
   public retries = { count: 0, delayMs: 0 };
 
   constructor(config: CosmosClientConfig) {
-    this.endpoint = config.endpoint;
-    this.signer = getSigner(config.masterKey);
+    if (config.connectionString) {
```

Comment:
mind adding a client.spec.ts with coverage for this constructor branching logic? If not no worries, I can add it later.

Reply:
I added a test, although I wasn't able to actually test it because I need  the proper test setup and a connection string. Let me know if this works :)